protobufs/protocol/protocol.proto

```diff
@@ -0,0 +1,24 @@
+syntax = "proto3";
+
+package protocol;
+
+option go_package = "protocolpb";
+option java_multiple_files = true;
+option java_outer_classname = "ProtocolProto";
+option java_package = "com.protocol";
+
+enum MessageType {
+  MESSAGE_TYPE_INVALID = 0;
+  MESSAGE_TYPE_REQUEST_SIGNATURE = 1;
+  MESSAGE_TYPE_REQUEST_SIGNATURE_REP = 2;
+  MESSAGE_TYPE_SEND_ANCHORED_DOC = 3;
+  MESSAGE_TYPE_SEND_ANCHORED_DOC_REP = 4;
+  MESSAGE_TYPE_ERROR = 5;
```

Comment:
Invalid and Error shouldn't be part of the Type. They are unexpected errors.

IMO, we should not tie the Type in the protocol itself.

Reply:
That is where we use error in the as part of the Message. IMO Error or invalid Type are not same as `MESSAGE_TYPE_SEND_ANCHORED_DOC`.